src/olympia/devhub/tasks.py

```diff
@@ -110,9 +110,9 @@ def create_version_for_upload(addon, upload,
         # review since listed and sideload aren't supported for creation yet.
         if addon.status == amo.STATUS_NULL:
             if disallow_preliminary_review:
-                addon.update(status=amo.STATUS_PUBLIC)
+                addon.update(status=amo.STATUS_NOMINATED)
```

Comment:
So, hold on. Does that mean that, without this patch, we set add-ons as public without a review somehow ?

In addition, if that fixes your bug, then why was the issue only happening in the API ?